ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc

```diff
@@ -0,0 +1,421 @@
+    }
+
+    ieTensorTypeStr = eachInputInfo->getPrecision().name();
+    nnsTensorType = TensorFilterOpenvino::convertFromIETypeStr(ieTensorTypeStr);
+    if (nnsTensorType == _NNS_END) {
+      g_critical ("The type of tensor elements, %s, "
+          "in the model is not supported", ieTensorTypeStr.c_str());
+      goto failed;
+    }
+
+    info->info[i].type = nnsTensorType;
+    info->info[i].name = g_strdup(eachInputInfo->name().c_str());
+  }
+
+  return TensorFilterOpenvino::RetSuccess;
+
+failed:
+  g_critical ("Failed to get dimension information about input tensor");
+
+  return ret;
+}
+
+/**
+ * @brief	Get the information about the dimensions of output tensors from the given model
+ * @param[out] info metadata containing the dimesions and types information of the output tensors
+ * @return 0 (TensorFilterOpenvino::RetSuccess) if OK, negative values if error
+ */
+int
+TensorFilterOpenvino::getOutputTensorDim (GstTensorsInfo * info)
+{
+  InferenceEngine::OutputsDataMap *outputsDataMap = &(this->_outputsDataMap);
+  InferenceEngine::OutputsDataMap::iterator outputDataMapIter;
+  int ret, i, j;
+
+  gst_tensors_info_init (info);
+
+  info->num_tensors = (uint32_t) outputsDataMap->size();
+  if (info->num_tensors > NNS_TENSOR_SIZE_LIMIT) {
+    g_critical ("The number of output tenosrs in the model "
+        "exceeds more than NNS_TENSOR_SIZE_LIMIT, %s",
+        NNS_TENSOR_SIZE_LIMIT_STR);
+    ret = RetEOverFlow;
+    goto failed;
+  }
+
+  for (outputDataMapIter = outputsDataMap->begin (), i = 0;
+      outputDataMapIter != outputsDataMap->end (); ++outputDataMapIter, ++i) {
+    InferenceEngine::SizeVector::reverse_iterator sizeVecRIter;
+    InferenceEngine::TensorDesc eachOutputTenserDesc;
+    InferenceEngine::SizeVector dimsSizeVec;
+    InferenceEngine::DataPtr eachOutputInfo;
+    std::string ieTensorTypeStr;
+    tensor_type nnsTensorType;
+
+    eachOutputInfo = outputDataMapIter->second;
+    eachOutputTenserDesc = eachOutputInfo->getTensorDesc ();
+    dimsSizeVec = eachOutputTenserDesc.getDims ();
+    if (dimsSizeVec.size () > NNS_TENSOR_RANK_LIMIT) {
+      g_critical ("The ranks of dimensions of OutputTensor[%d] in the model "
+          "exceeds NNS_TENSOR_RANK_LIMIT, %u", i, NNS_TENSOR_RANK_LIMIT);
+      ret = RetEOverFlow;
+      goto failed;
+    }
+
+    for (sizeVecRIter = dimsSizeVec.rbegin(), j = 0;
+        sizeVecRIter != dimsSizeVec.rend (); ++sizeVecRIter, ++j) {
+      info->info[i].dimension[j] = (*sizeVecRIter != 0 ? *sizeVecRIter : 1);
+    }
+    for (int k = j; k < NNS_TENSOR_RANK_LIMIT; ++k) {
+      info->info[i].dimension[k] = 1;
+    }
+
+    ieTensorTypeStr = eachOutputInfo->getPrecision ().name ();
+    nnsTensorType = TensorFilterOpenvino::convertFromIETypeStr (ieTensorTypeStr);
+    if (nnsTensorType == _NNS_END) {
+      g_critical ("The type of tensor elements, %s, "
+          "in the model is not supported", ieTensorTypeStr.c_str());
+      goto failed;
```

Comment:
I think you may need to set `ret` before performing this `goto` statement.

Reply:
@dongju-chae Fixed. Thanks.